app/client/jsx/Map.jsx

```diff
@@ -4,51 +4,23 @@ import { connect } from 'react-redux'
 import reducers from './reducers.jsx'
 import MapVisualization from './MapVisualization.js'
 import MapRoomInfo from './MapRoomInfo.jsx'
-import { HttpApi } from './WebAPI.jsx'
 
 class Map extends Component {
     constructor(props) {
         super(props)
-
         this.state = {
             highlighted: null,
             users: []
         }
-
-        this.httpApi = new HttpApi()
-
-        this.props.socketApi.on('user-left-room', this.fetchUsers.bind(this))
-        this.props.socketApi.on('user-entered-room', this.fetchUsers.bind(this))
-        this.props.socketApi.on('user-disconnected', this.fetchUsers.bind(this))
     }
 
-    async fetchUsers() {
```

Comment:
Instead of sending an HTTP request to get all the users, why not send all the users down as part of the socket event? Then you can avoid an extra round trip to the server (and possibly some database queries as well).